### DISCUSSION TITLE
Özel Sayılar

### DISCUSSION THREAD

**Discussion by OmerAkbayrak** [sonsuzus/sonsuzus#407] [Soru Cevap]
### Problem
Bir doğal sayının özel olması için sadece 4 ve 7 rakamlarından oluşması gerekir. 1<x<y<N şeklinde olacak şekilde kaç tane asal özel sayı çifti(x,y) olduğunu bulunuz.
### Girdi Formatı:
İlk satır N değişkeni için bir sayı içerecek.(1<N<10⁹)
### Çıktı Formatı:
Sorunun cevabını gösteren tek bir sayı yazdırılacak.

### _Örnek Çıktı: 1_

> 
> ### Açıklama:
> _**Açıklama Linki: https://en.wikipedia.org/wiki/Coprime_integers**_
> Cevap : 1 Çünkü 4 ve 7, 10'dan küçük ve bu iki sayı bir çift ediyor.
>

> **_Soru Linki https://www.hackerearth.com/practice/basic-programming/implementation/basics-of-implementation/practice-problems/algorithm/happy-numbers-4a67748e/_**

**Comment by EfeMertIrez** [comment]
```
from math import gcd
 
 
def define_special_numbers(mx, vals, x=0):
    if x > mx:
        return
    if x > 0:
        vals.append(x)
    define_special_numbers(mx, vals, 10 * x + 4)
    define_special_numbers(mx, vals, 10 * x + 7)
 
 
n = int(input())
special_numbers = []
define_special_numbers(n, special_numbers)
count = k = 0
for i in special_numbers:
    k += 1
    for j in special_numbers[k:]:
        if gcd(i, j) == 1:
            count += 1
```
